src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -97,24 +109,83 @@ public class MaturedBlocksSyncManager : IMaturedBlocksSyncManager
             this.initialBlockDownloadState = initialBlockDownloadState;
             this.nodeLifetime = nodeLifetime;
             this.conversionRequestRepository = conversionRequestRepository;
+            this.counterChainSettings = counterChainSettings;
+            this.httpClientFactory = httpClientFactory;
             this.chainIndexer = chainIndexer;
             this.externalApiPoller = externalApiPoller;
             this.conversionRequestFeeService = conversionRequestFeeService;
             this.network = network;
             this.federationManager = federationManager;
+            this.mainChainActivationHeight = int.MaxValue;
 
             this.lockObject = new object();
             this.logger = LogManager.GetCurrentClassLogger();
         }
 
+        public class ConsensusClient : RestApiClientBase
+        {
+            /// <summary>
+            /// Currently the <paramref name="url"/> is required as it needs to be configurable for testing.
+            /// <para>
+            /// In a production/live scenario the sidechain and mainnet federation nodes should run on the same machine.
+            /// </para>
+            /// </summary>
+            public ConsensusClient(ICounterChainSettings counterChainSettings, IHttpClientFactory httpClientFactory)
+                : base(httpClientFactory, counterChainSettings.CounterChainApiPort, "Consensus", $"http://{counterChainSettings.CounterChainApiHost}")
+            {
+            }
+
+            public Task<List<ThresholdActivationModel>> GetLockedInDeployments(CancellationToken cancellation = default)
+            {
+                return this.SendGetRequestAsync<List<ThresholdActivationModel>>("lockedindeployments", null, cancellation);
+            }
+        }
+
+        public void RecordCounterChainActivations()
+        {
+            // If this is the main chain then ask the side-chain for its activation height.
+            if (this.network.Name.StartsWith("Cirrus"))
+                return;
+
+            if (this.mainChainActivationHeight != int.MaxValue)
+                return;
+
+            ConsensusClient consensusClient = new ConsensusClient(this.counterChainSettings, this.httpClientFactory);
+            List<ThresholdActivationModel> lockedInActivations = consensusClient.GetLockedInDeployments(this.nodeLifetime.ApplicationStopping).ConfigureAwait(false).GetAwaiter().GetResult();
+            if (lockedInActivations == null || lockedInActivations.Count == 0)
+                return;
```

Comment:
We need to add some debug logging so that we know when something goes wrong during tresting. Please add a debug log here saying that nothing got returned.

Reply:
Done.